buildhub/settings.py

```diff
@@ -288,6 +288,29 @@ def LOGGING(self):
             },
         }
 
+    # Security related
+
+    SILENCED_SYSTEM_CHECKS = values.ListValue(
+        [
+            # This rule is about having "X-FRAME-OPTIONS" set to "DENY" which is
+            # not important for Buildhub2 because it's considered NOT dangerous to
+            # put any of this data inside an iframe.
+            "security.W019",
+            # There are no POST endpoints that *change* data that is exposed.
+            # I.e. you can't CSRF attack this site because there's no URL to do so
+            # with.
+            "security.W003",
+        ]
+    )
```

Comment:
@g-k Can you review this?

Basically, I'm making a judgment that it is safe to include any Buildhub URL inside and iframe. Also, same for CSRF since there is no endpoint that uses HTTP post and authentication to make a database write. 

Let me know if you *can't* review this or if you need a bit more context as to what the hell this context is.

Reply:
@peterbe It looks this service is tightly scoped to serve static content and fetch and update more content from trusted sources, so this seems reasonable to me. We might need to update the ZAP baseline scan.

Some additional context from the questions below would be helpful:

* Do we have to disable these warnings at the service level? Will nginx add them back (if so, that might be worth mentioning in config comments)?

* How slow is the search endpoint? CSRF for the right queries might result in DoS, but that's probably an acceptable risk if we have autoscaling or the API is unauthed anyway.

* Do we plan to add any endpoints that modify data and could be vulnerable to CSRF?

* If someone injected malicious data (via taskcluster or SQS), would buildhub eventually serve it? Do you have examples of current data we're serving or know how it's validated?

* What's are the use cases for the content buildhub currently serves and for framing that content?

Reply:
> Do we have to disable these warnings at the service level? Will nginx add them back.

```
$ curl -v https://buildhub.moz.tools
< HTTP/2 200
< server: nginx/1.15.4
< x-frame-options: SAMEORIGIN
```

So for x-frame-options, our Nginx does not already set this. That default `x-frame-options: SAMEORIGIN` comes from Django and is pretty sane. 

And regarding CSRF cookies, Nginx does not get involved in that. 

So I think nothing needs to be done for these two above mentioned exceptions. 


> How slow is the search endpoint? CSRF for the right queries might result in DoS

It's reasonably fast after all. It's just Elasticsearch queries and all around it involves no writes. 
It's not an option to try to cache the results because if someone wanted to bypass any caching they could too easily just change the JSON they POST and then the hashing of a cache key would be moot. 

The right thing to do would be to a rate-limiter which is pretty easy to do with Django. It relies on a mix of IP and user-agent I think. 

Either way, it's very unlikely to happen because the service is kinda "boring" and there's so little to gain by DDoSing it. 

> Do we plan to add any endpoints that modify data and could be vulnerable to CSRF?

No, and the day we do add any such things I would definitely add CSRF. However, there is the "human element" of one day adding a POST endpoint that depends on auth and does writes and perhaps the developer who adds that forgets to enable CSRF again. Short answer; no there are no such plans. 

> If someone injected malicious data (via taskcluster or SQS), would buildhub eventually serve it?

All content from the server is rendered via a React app that always and everywhere escapes strings when displaying them in the DOM. I have no formal proof or experiment to assert this but I feel confident enough. 

> What's are the use cases for the content buildhub currently serves and for framing that content?

Honestly, none. But if someone does at least they won't be able to do an iframe with 100% and 100% height under a different and rogue domain name (because we still default to x-frame-options: SAMEORIGIN).

Reply:
Thanks for filling me in @peterbe! lgtm

Reply:
I think we're good enough to land this. Discussing it helped me form my own, stronger, opinion about it. If it's not perfect it's fine since the bar is so low for this project.